daxgen/daxgen.py

```diff
@@ -0,0 +1,184 @@
+import networkx as nx
+from Pegasus.DAX3 import ADAG, File, Job, Link
+
+
+# Set of attributes unique to file nodes. Do not have to be complete,
+_FILE_ATTRS = {'lfn', 'pfns'}
+
+
+class Daxgen(object):
+    """Generator of Pegasus DAXes.
+
+    Parameters
+    ----------
+    graph : `networkx.DiGraph`, optional
+        Graph representation of the workflow, defaults to an empty graph.
+    """
+
+    def __init__(self, graph=None):
+        self.g = graph if graph is not None else nx.DiGraph()
```

Comment:
I don't believe there's a specific rule in the coding standard against single letter variable names in class instance variables, but you might want to consider renaming "self.g" to something more descriptive to make it clearer.